src/NServiceBus.Testing/TestingContext.cs

```diff
@@ -49,25 +47,22 @@ public void Validate()
             }
         }
 
-        static TimeSpan GetWithin(SendOptions options)
+        static TimeoutMessage<object> GetTimeoutMessage(object message, SendOptions options)
         {
             var within = options.GetDeliveryDelay();
-
-            if (!within.HasValue)
+            if (within.HasValue)
             {
-                var dateTimeOffset = options.GetDeliveryDate();
-                if (dateTimeOffset != null)
-                {
-                    within = dateTimeOffset.Value - DateTimeOffset.Now;
-                }
+                return new TimeoutMessage<object>(message, options, within.Value);
             }
 
-            if (!within.HasValue)
+            var dateTimeOffset = options.GetDeliveryDate();
+            if (dateTimeOffset.HasValue)
             {
-                throw new Exception("No time has been set for the timeout message");
+                return new TimeoutMessage<object>(message, options, dateTimeOffset.Value);
             }
 
-            return within.Value;
+
+            throw new Exception("No TimeSpan or DateTime has been set for the timeout message");
```

Comment:
Should we change the testing FX to not look at the header to determine if the message is a timeout out but instead require there to be a `options.GetDeliveryDelay()` since that would clean up the code?